lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1191,6 +1201,213 @@ impl PrerenderedU8Function {
     }
 }
 
+struct TreeGrid {
+    grid_width: u32,
+    grid_height: u32,
+    offset_x: i32,
+    offset_y: i32,
+    chunk_size: u32,
+    rtrees: Vec<RwLock<RTree<[i32; 2]>>>,
+}
+
+// This is a grid of rtrees
+// The idea is that most pixels after the first couple steps will have their neighbors close by
+impl TreeGrid {
+    pub fn new(width: u32, height: u32, chunk_size: u32, offset_x: u32, offset_y: u32) -> TreeGrid {
+        let mut rtrees: Vec<RwLock<RTree<[i32; 2]>>> = Vec::new();
+        let grid_width = width / chunk_size + 1;
+        let grid_height = height / chunk_size + 1;
```

Comment:
If you want round-up behavior here, it should probably be written as `(a + (b - 1)) / b`, e.g. `(width + chunk_size - 1) / chunk_size`. With `a / b + 1`, you get an extra element when `a` is divisible by `b`.

Reply:
great suggestion will do!